src/types/dependency-container.ts

```diff
@@ -39,7 +39,9 @@ export default interface DependencyContainer {
     instance: T
   ): DependencyContainer;
   resolve<T>(token: InjectionToken<T>): T;
+  resolve<T>(token: InjectionToken<T>, parent?: constructor<any>): T;
```

Comment:
I think you can remove the optionality of the `parent` param since we have the above overload. (same with `resolveAll(...)`)

Reply:
I removed the optionality, and updated `predicateAwareClassFactory` to match.